core/CPUFactory.cpp

```diff
@@ -94,6 +94,10 @@ auto olympia::CPUFactory::bindTree_(sparta::RootTreeNode* root_node,
                 setTLB(*private_nodes_.at(num_of_cores)->getResourceAs<olympia::SimpleTLB>());
         (core_tree_node->getChild("preloader")->getResourceAs<olympia::Preloader>())->
             preload();
+        auto rob = core_tree_node->getChild("rob")->getResourceAs<olympia::ROB>()->getContainer();
+        auto lsu = core_tree_node->getChild("lsu")->getResourceAs<olympia::LSU>();
+        rob->registerForNotification<bool, LSU, &LSU::onRobDrained_>
+            (lsu, "rob_notif_channel");
```

Comment:
Go ahead and use the MACRO for this:
```suggestion
        rob->REGISTER_FOR_NOTIFICATION(onRobDrained_, bool, "rob_notif_channel");
```

Reply:
Sounds good

Reply:
@knute-sifive Looks like the macro needs to be defined in the class as it references (this)